collect_app/src/main/java/org/odk/collect/android/database/helpers/FormsDatabaseHelper.java

```diff
@@ -125,7 +128,7 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
     public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         try {
             SQLiteUtils.dropTable(db, FORMS_TABLE_NAME);
-            createFormsTableV8(db);
+            createFormsTableV9(db);
```

Comment:
Previously, all of the data in the forms table could be recreated from a form definition except for the last detected hash. Now delete adds more state that can't be recreated. Still, the consequences of "undeleting" forms on downgrade are not very dire and we don't expect downgrade to be common so let's stick with this and not worry about it. (This is just a note to make the decision explicit to our future selves)